pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -68,12 +68,14 @@ type AzVolumeAttachment struct {
 
 // AzVolumeAttachmentSpec is the spec for a AzVolumeAttachment resource
 type AzVolumeAttachmentSpec struct {
-	UnderlyingVolume string `json:"underlyingVolume"`
+	AzVolumeName     string `json:"underlyingVolume"`
```

Comment:
Go and JSON field names should be consistent. I would suggest using pkg pubilc versions of the JSON filed names, e.g. `UnderlyingVolume` here.